src/dut_power.rs

```diff
@@ -306,9 +308,11 @@ impl DutPwrThread {
         // open drain.
         // The outputs on later TACs should however be driven open drain
         // for EMI reasons.
-        let flags = match pwr_line.chip().label() {
-            "pca9570" => LineRequestFlags::OUTPUT,
-            _ => LineRequestFlags::OUTPUT | LineRequestFlags::OPEN_DRAIN,
+        let flags = match hardware_generation {
+            HardwareGeneration::Gen1 => LineRequestFlags::OUTPUT,
+            HardwareGeneration::Gen2 | HardwareGeneration::Gen3 => {
+                LineRequestFlags::OUTPUT | LineRequestFlags::OPEN_DRAIN
+            }
```

Comment:
This should be moved into a method of `HardwareGeneration` instead of matching here.